retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaCallFactory.java

```diff
@@ -129,80 +135,55 @@ public Request request() {
             return request;
         }
 
-        private synchronized CompletableFuture<AggregatedHttpMessage> createRequest() {
-            if (executed) {
-                throw new IllegalStateException("Already Executed");
+        private synchronized void createRequest() {
+            if (httpResponse != null) {
+                throw new IllegalStateException("executed already");
             }
-            final CompletableFuture<AggregatedHttpMessage> future = doCall(callFactory.httpClient, request)
-                    .aggregate();
-            executed = true;
-            return future;
+            executionStateUpdater.compareAndSet(this, ExecutionState.IDLE, ExecutionState.RUNNING);
```

Comment:
I tried but this will change behavior of retrofit, this test(https://github.com/line/armeria/pull/354/files#diff-ccc735bd0bfc4be834afe1ca42488d8cR275) will be failed. Default retrofit is allowed to cancel then enqueue.